src/pip/_internal/utils/subprocess.py

```diff
@@ -188,7 +185,8 @@ def call_subprocess(
             stderr=subprocess.STDOUT, stdin=subprocess.PIPE,
             stdout=subprocess.PIPE, cwd=cwd, env=env,
         )
-        proc.stdin.close()
+        if proc.stdin:
+            proc.stdin.close()
```

Comment:
```suggestion
        assert proc.stdin
        assert proc.stdout
        proc.stdin.close()
```

Reply:
Fixed in https://github.com/pypa/pip/pull/8322/commits/a847ddab25a2502dc9f0402347c83c8d9955c022